[REVIEW] Added RAPIDS workflow on AKS with necessary setup guides and notebooks (#118)
rapidsai/cloud-ml-examples: azure/notebooks/kubernetes/Detailed_setup_guide.md

# [Detailed Guide to use Dask on Azure Kubernetes Service (AKS)](#anchor-start)

> For all the next steps, we will be using the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli), however the same can be achieved through the Azure GUI Portal. 

Reviewer: ```suggestion
For all the next steps, we will be using the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli), however the same can be achieved through the Azure Portal. 
```

Reviewer: Can we also link the portal here

### [Step 0: Install and authenticate with Azure CLI](#anchor-install-azurecli)
- Install the `az` cli using 
    ```
    curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
    ````

>     in the computer from where you will be running everythning from. Get rid of the `sudo` if you are running things from inside a container. 

Reviewer: ```suggestion
    on the computer from where you will be running these examples from. You can remove `sudo` if running inside a Docker container. 
```

> - Once `az` is installed, make sure you you configure the local `az` cli to work with your Azure credentials, do `az login` and authenticate from Microsoft's website.

Reviewer: ```suggestion
- Once `az` is installed, make sure you configure the local `az` cli to work with your Azure credentials, run `az login` and authenticate from Microsoft's website.
```

- For more details follow the steps [Here](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli). 

### [Step 1: Install Kubectl](#anchor-install-kubectl)

- Install `kubectl` to access your cluster from the local machine from the following link : [https://kubernetes.io/docs/tasks/tools/] depending on your operating system. 

### [Step 2: Set some environment variables](#anchor-set-env-variables)

> We will set some environment variables beforehand, namely to help us deploy some resources quickly.  We will keep on using some common values for the rest of the deployments. . 

Reviewer: ```suggestion
We will set some environment variables beforehand, namely to help us deploy some resources quickly.  We will continue using common values for the rest of the deployments. . 
```

```bash
REGION_NAME=<your preferred location>
RESOURCE_GROUP=<your preferred resource group name>
SUBNET_NAME=<optional subnet name>
VNET_NAME=<optional vnet name>
VERSION=$(az aks get-versions --location $REGION_NAME --query \
                'orchestrators[?!isPreview] | [-1].orchestratorVersion' \--output tsv)
AKS_CLUSTER_NAME=<your cluster name>
VM_SIZE=Standard_NC12s_v3 # or any other VM size. We use VMs with GPU
```

- We would need a resource group. You can create a resource group in Azure `az group create  --name $RESOURCE_GROUP --location $REGION_NAME`, or use an existing one.

> - Secondly, we get the latest non preview Kubernetes version and store it in an env variable using. This provides that the latest version is 1.20.5. Optionally you can directly set the Kubernetes version in the environment variable `VERSION`.

Reviewer: Were you planning to add anything after using?

Author: This is a good catch. Left over from restructuring. Fixed.

> - **NOTE 1**: There are two network modes to choose from when deploying an AKS cluster. The default one is *Kubenet networking* which we will use here. The other more sophisticated one is Azure Container Networking Interface. We will not need this at the moment. Please look [here](https://docs.microsoft.com/en-us/learn/modules/aks-workshop/02-deploy-aks) to get an idea about how to deploy Azure CNI.  

Reviewer: ```suggestion
- **NOTE 1**: There are two network modes to choose from when deploying an AKS cluster. The default one is *Kubenet networking* which we will use here. 
```

> - **NOTE 2**: Depending on your account limitations, the number and type of VMs that you can spin up may vary. Also there may be zone limitations. Make sure you spin up VMs having GPU capabilities GPU: NVIDIA Pascal™ or better with compute capability 6.0+. 

Reviewer: ```suggestion
- **NOTE 2**: Depending on your account limitations, the number and type of VMs that you can spin up may vary. Also there may be zone limitations. Make sure you spin up VMs with GPUs: NVIDIA Pascal™ or better with compute capability 6.0+. 
```

Reviewer: Can you add which VMs on Azure? Or the ones that you used for these examples?

Author: Added

> ### [Step 3: Create the cluster and get credentials](#anchor-create-aks-cluster)

Reviewer: ```suggestion
### [Step 3: Create the cluster with credentials](#anchor-create-aks-cluster)
```

Author: I actually mean getting the kubernetes credentials _after_ I create the cluster. I think this is fine.

Once you verify that you are allowed to use the necessary VM sizes in your preferred location, now its time to create a managed kubernetes cluster, namely an AKS cluster. The process is pretty simple. Also, after you successfully deploy a cluster with a node-pool of some nodes, you will be able to run workers as pods on the kubernetes cluster using [dask-kubernetes](https://github.com/dask/dask-kubernetes).

> - Anyway, first, run the following command to create a AKKS cluster running the latest kubernetes version. It will take a few minutes before it completes. Grab a coffee :coffee: :coffee: .

Reviewer: ```suggestion
- Let's first run the following command to create a AKS cluster using the latest kubernetes version. It will take a few minutes before it completes. Grab a coffee :coffee: :coffee: .
```
